File: src/cz/vutbr/fit/interlockSim/sim/PathSeparatorChangeException.java
/* Brno University of Technology
 * Faculty of Information Technology
 * 
 * BSc Thesis	2006/2007
 * 
 * Railway Interlocking Simulator
 * 
 * Bedrich Hovorka
 */
package cz.vutbr.fit.interlockSim.sim;

import cz.vutbr.fit.interlockSim.objects.paths.PathSeparator;

/**
 * @author beda
 *
 */
public class PathSeparatorChangeException extends SimulationException {
	/**
	 * @param object
	 */
	public PathSeparatorChangeException(PathSeparator object) {
		super(object);
	}

	/**
	 * @param message
	 * @param object
	 */
	public PathSeparatorChangeException(String message, PathSeparator object) {
		super(message, object);
	}

	/**
	 * @param cause
	 * @param object
	 */
	public PathSeparatorChangeException(Throwable cause, PathSeparator object) {
		super(cause, object);
	}

	/**
	 * @param message
	 * @param cause
	 * @param object
	 */
	public PathSeparatorChangeException(String message, Throwable cause, PathSeparator object) {
		super(message, cause, object);
	}
	
	@Override
	public PathSeparator getObject() {
		return (PathSeparator) super.getObject();
	}
}

File: src/main/kotlin/cz/vutbr/fit/interlockSim/exceptions/PathSeparatorChangeException.kt
/* Brno University of Technology
 * Faculty of Information Technology
 *
 * BSc Thesis  2006/2007
 *
 * Railway Interlocking Simulator
 *
 * Bedrich Hovorka
 */
package cz.vutbr.fit.interlockSim.exceptions

import cz.vutbr.fit.interlockSim.objects.paths.PathSeparator

/**
 * @author beda
 *
 */
class PathSeparatorChangeException : SimulationException {
	/**
	 * @param object
	 */
	constructor(obj: PathSeparator) : super(obj)

	/**
	 * @param message
	 * @param object
	 */
	constructor(message: String, obj: PathSeparator) : super(message, obj)

	/**
	 * @param cause
	 * @param object
	 */
	constructor(cause: Throwable, obj: PathSeparator) : super(cause, obj)

	/**
	 * @param message
	 * @param cause
	 * @param object
	 */
	constructor(message: String, cause: Throwable, obj: PathSeparator) : super(message, cause, obj)

	override fun getObject(): PathSeparator? = super.getObject() as? PathSeparator
}
